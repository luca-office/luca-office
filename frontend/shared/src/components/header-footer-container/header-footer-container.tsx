import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {
  backgroundColorBright,
  borderRadius,
  cardBottomColor,
  CustomStyle,
  Flex,
  headerBoxShadow,
  spacing,
  spacingLarge,
  spacingMedium
} from "../../styles"

export interface HeaderFooterContainerProps {
  readonly customContentContainerStyles?: CSSInterpolation
}

export const HeaderFooterContainer: React.FC<React.PropsWithChildren<HeaderFooterContainerProps & CustomStyle>> = ({
  children,
  customStyles,
  customContentContainerStyles
}) => {
  return (
    <div css={[styles.headerContainer, customStyles]}>
      <div css={styles.headerContainerHeader} className={"header-bar"} />
      <div css={[styles.containerContent, customContentContainerStyles]} className={"content-container"}>
        {children}
      </div>
      <div css={styles.headerContainerFooter} className={"footer-bar"} />
    </div>
  )
}
const styles = {
  headerContainer: css({
    backgroundColor: backgroundColorBright
  }),
  headerContainerHeader: css({
    height: spacingMedium,
    borderRadius: spacing(borderRadius, 0),
    backgroundColor: cardBottomColor,
    boxShadow: headerBoxShadow
  }),
  headerContainerFooter: css({
    height: spacingMedium,
    backgroundColor: cardBottomColor
  }),
  containerContent: css(Flex.column, {
    padding: spacingLarge
  })
}
