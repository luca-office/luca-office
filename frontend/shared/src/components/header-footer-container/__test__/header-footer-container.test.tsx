import {render} from "@testing-library/react"
import {mount, shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Text} from "../.."
import {HeaderFooterContainer} from "../header-footer-container"

describe("Empty Detail View", () => {
  it("renders correctly", () => {
    const component = create(<HeaderFooterContainer />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders structure correctly", () => {
    const component = shallow(
      <HeaderFooterContainer>
        <Text>Test</Text>
      </HeaderFooterContainer>
    )

    expect(component.find(Text)).toHaveLength(1)

    expect(component.find(".header-bar")).toHaveLength(1)
    expect(component.find(".content-container")).toHaveLength(1)
    expect(component.find(".footer-bar")).toHaveLength(1)
  })
})
