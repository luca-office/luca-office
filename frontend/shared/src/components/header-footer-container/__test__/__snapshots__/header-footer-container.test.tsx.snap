// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Detail View renders correctly 1`] = `
<div
  className="css-wxyiz4-HeaderFooterContainer"
>
  <div
    className="header-bar css-1kk0xzt-HeaderFooterContainer"
  />
  <div
    className="content-container css-na1jjh-HeaderFooterContainer"
  />
  <div
    className="footer-bar css-ji67lt-HeaderFooterContainer"
  />
</div>
`;
